crates/admin/src/cluster_controller/service.rs

```diff
@@ -568,53 +454,178 @@ impl<T: TransportConnect> Service<T> {
-                    Duration::from_secs(2),
-                    self.update_cluster_configuration(
-                        partition_replication,
-                        default_provider,
-                        num_partitions,
-                    ),
-                )
-                .await
-                {
-                    Ok(result) => {
-                        let _ = response_tx.send(result);
-                    }
-                    Err(_timeout) => {
-                        let _ =
-                            response_tx.send(Err(anyhow!("Timeout on writing to metadata store")));
-                    }
-                }
+                let metadata_writer = self.metadata_writer.clone();
+
+                // receiver will get error if response_tx is dropped
+                let _ =
+                    TaskCenter::spawn(TaskKind::Disposable, "update-cluster-config", async move {
+                        match tokio::time::timeout(
+                            Duration::from_secs(2),
+                            update_cluster_configuration(
+                                metadata_writer,
+                                partition_replication,
+                                default_provider,
+                                num_partitions,
+                            ),
+                        )
+                        .await
+                        {
+                            Ok(result) => {
+                                let _ = response_tx.send(result);
+                            }
+                            Err(_timeout) => {
+                                let _ = response_tx.send(Err(anyhow!(
+                                    "Timeout writing updated configuration to metadata store"
+                                )));
+                            }
+                        }
+
+                        Ok(())
+                    });
             }
             ClusterControllerCommand::SealAndExtendChain {
                 log_id,
                 min_version,
                 extension,
                 response_tx,
-            } => self.seal_and_extend_chain(log_id, min_version, extension, response_tx),
+            } => {
+                let bifrost = self.bifrost.clone();
+                let observed_cluster_state = self.observed_cluster_state.clone();
+
+                // receiver will get error if response_tx is dropped
+                _ = TaskCenter::spawn(TaskKind::Disposable, "seal-and-extend", async move {
+                    let result = SealAndExtendTask {
+                        log_id,
+                        extension,
+                        min_version,
+                        bifrost,
+                        observed_cluster_state,
+                    }
+                    .run()
+                    .await;
+
+                    _ = response_tx.send(result);
+                    Ok(())
+                });
+            }
```

Comment:
This is just inlining `self.seal_and_extend_chain()` and not changing any logic, right? If yes, what's the motivation for inlining it?

Reply:
This is indeed! I just wanted to make it more in line with the other arms of this select - I felt it was helpful to make it obvious at a glance what these are doing. They're not super long so it seemed like a reasonable tradeoff. Happy to move it back to how it was of course.